### PR TITLE
feat(theme-default): enhance color transitions

### PR DESCRIPTION
**Pull request by Afanyiyu**
like #287

### CODE REVIEW

File: packages/@vuepress/theme-default/src/client/styles/normalize.scss [hunk 1]
```diff
@@ -48,9 +48,7 @@ code {
   background-color: var(--c-bg-light);
   border-radius: 3px;
   overflow-wrap: break-word;
-  transition:
-    background-color ease 0.3s,
-    color ease 0.3s;
+  transition: background-color ease 0.3s, color ease 0.3s;
```
Comment (Afanyiyu) [inline packages/@vuepress/theme-default/src/client/styles/normalize.scss:51]: lint

File: packages/@vuepress/theme-default/src/client/styles/normalize.scss [hunk 2]
```diff
@@ -96,6 +94,7 @@ h2 {
   font-size: 1.65rem;
   padding-bottom: 0.3rem;
   border-bottom: 1px solid var(--c-border);
+  transition: border-bottom-color ease 0.3s;
```
Comment (Afanyiyu) [inline packages/@vuepress/theme-default/src/client/styles/normalize.scss:97]: Add `transition` for `border-bottom` in `h2`

File: packages/@vuepress/theme-default/src/client/styles/sidebar.scss [hunk 3]
```diff
@@ -14,6 +14,7 @@
   .navbar-links {
     display: none;
     border-bottom: 1px solid var(--c-border);
+    transition: border-bottom-color ease 0.3s;
```
Comment (Afanyiyu) [inline packages/@vuepress/theme-default/src/client/styles/sidebar.scss:17]: Add `transition` for `border-bottom` in `navbar-links`

File: packages/@vuepress/plugin-search/src/client/styles/search.css [hunk 4]
```diff
@@ -16,7 +16,7 @@
   line-height: 2rem;
   padding: 0 0.5rem 0 2rem;
   outline: none;
-  transition: all 0.2s ease;
+  transition: all ease 0.3s;
```
Comment (Afanyiyu) [inline packages/@vuepress/plugin-search/src/client/styles/search.css:19]: Use `ease 0.3s` like other `transition`

File: packages/@vuepress/theme-default/src/client/styles/layout.scss [hunk 5]
```diff
@@ -160,7 +161,6 @@ h6 {
     top: 0;
     padding-top: var(--navbar-height);
     transform: translateX(-100%);
-    transition: transform 0.2s ease;
```
Comment (Afanyiyu) [inline packages/@vuepress/theme-default/src/client/styles/layout.scss:163]: Removed here and moved to [L37](https://github.com/vuepress/vuepress-next/pull/321/files#diff-4dd3e0f55ac29286ecca1b5abe7786ded54987ea3057f5bcd7cf4498823b8734R37)

This `transition` in `@media` will override the normal `transition` for `background-color` and `border-right-color`

**Review by Shigma — COMMENTED** [review]
maybe we can add a `--c-transition` or `--t-color` (t stands for transition) for all these transitions, so that users can decide by themselves:

```css
:root {
  --t-color: color 0.3s ease, border-color 0.3s ease, background-color 0.3s ease;
}

.some-element {
  transition: var(--t-color);
}
```

**Comment by Shigma** [comment]
Maybe it's better to place **_duration_** before **_timing function_** according to specification.

```css
/* Apply to 1 property */
/* property name | duration */
transition: margin-right 4s;

/* property name | duration | delay */
transition: margin-right 4s 1s;

/* property name | duration | timing function */
transition: margin-right 4s ease-in-out;

/* property name | duration | timing function | delay */
transition: margin-right 4s ease-in-out 1s;

/* Apply to 2 properties */
transition: margin-right 4s, color 1s;

/* Apply to all changed properties */
transition: all 0.5s ease-out;

/* Global values */
transition: inherit;
transition: initial;
transition: unset;
```

Link:

- https://developer.mozilla.org/en-US/docs/Web/CSS/transition
- https://drafts.csswg.org/css-transitions/#transition-shorthand-property

**Comment by meteorlxy** [comment]
If the changes of this PR is good, we can merge it first, then refactor with your suggestions in another PR. @Shigma

**Comment by Afanyiyu** [comment]
那我先 Resolve Conflicts

**Review by Shigma — APPROVED** [review]

**Pull request merged**